jsx/DataTable.js

```diff
@@ -380,100 +390,63 @@ class DataTable extends Component {
         }
       }
     }
+
     let rows = [];
-    let curRow = [];
-    let index = this.getSortedRows();
-    let matchesFound = 0; // Keeps track of how many rows where displayed so far across all pages
-    let filteredRows = this.countFilteredRows();
+    let filteredRows = this.getFilteredRows();
+    let filteredCount = filteredRows.length;
+    let index = this.sortRows(filteredRows);
     let currentPageRow = (rowsPerPage * (this.state.page.number - 1));
     let filteredData = [];
-    let useKeyword = false;
 
     if (this.props.filter.keyword) {
       useKeyword = true;
     }
 
-    // Push rows to data table
-    for (let i = 0;
-         (i < this.props.data.length) && (rows.length < rowsPerPage);
+    // Format each cell for the data table.
+    for (let i = currentPageRow;
+         (i < filteredCount) && (rows.length < rowsPerPage);
          i++
     ) {
-      curRow = [];
-
-      // Counts filter matches
-      let filterMatchCount = 0;
-      let keywordMatch = 0;
-      let filterLength = 0;
-
-      // Iterates through headers to populate row columns
-      // with corresponding data
-      for (let j = 0; j < this.props.fields.length; j += 1) {
-        let data = 'Unknown';
-
-        // Set column data
-        if (this.props.data[index[i].RowIdx]) {
-          data = this.props.data[index[i].RowIdx][j];
-        }
-
-        if (this.props.fields[j].filter) {
-          if (this.hasFilterKeyword(this.props.fields[j].filter.name, data)) {
-            filterMatchCount++;
-            filteredData.push(this.props.data[index[i].RowIdx]);
-          }
-        }
+        let rowIndex = index[i].RowIdx;
+        let rowData = this.props.data[rowIndex];
+        let curRow = [];
+
+        // Iterates through headers to populate row columns
+        // with corresponding data
+        for (let j = 0; j < this.props.fields.length; j += 1) {
+            if (this.props.fields[j].show === false) {
+                continue;
+            }
 
-        if (useKeyword === true) {
-          filterLength = Object.keys(this.props.filter).length - 1;
-          if (this.hasFilterKeyword('keyword', data)) {
-            keywordMatch++;
-          }
-        } else {
-          filterLength = Object.keys(this.props.filter).length;
-        }
+            let celldata = rowData[j];
+            let cell = null;
 
-        let key = 'td_col_' + j;
 
-        // Get custom cell formatting if available
-        if (this.props.getFormattedCell) {
-          if (this.props.fields[j].show === false) {
-            data = null;
-          } else {
-            // create mapping between rowHeaders and rowData in a row Object
-            const row = {};
-            this.props.fields.forEach((field, k) => {
-              row[field.label] = this.props.data[index[i].RowIdx][k];
-            });
```

Comment:
I think the removal of this may be causing some problems. This section of the code removes data when it is not meant to be shown and creates a row indexed by column header, which is what `getFormattedCell` needs to format the columns.